### DISCUSSION TITLE
Aqara vibration sensor does not work properly after update to 1.40.2-1 version

### DISCUSSION THREAD

**Discussion by bytdoma** [Koenkk/zigbee2mqtt#24185] [Devices]
Hello,
after an upgrade to new version 1.40.2-1 in Homeassistant Aqara vibration sensor only shows values for vibration and strength, others: angle_x, angle_y, angle_z, x_axis, y_axis, z_axis became Unknown. I did not check other entities unfortunately. I reconfigured the device, prompted, restarted Homeassistant and let it run for about na hour, but no change. I downgraded back to 1.40.1-1 via my backup and it is working again.

**Comment by MarcoZ76** [comment]
I have the same issue (i have 2 devices, on both this issue)

**Comment by Maggie70** [comment]
Same problem here, i downgraded and it working again

**Comment by MurilloRacing** [comment]
Same problem here.

**Reply by MurilloRacing** [reply]
I downgraded to 1.40.1-1, still has same issue. Aqara vibration sensor: angle_x, angle_y, angle_z, x_axis, y_axis, z_axis Unknown

**Reply by Maggie70** [reply]
try shaking the sensor to trigger it

**Reply by MurilloRacing** [reply]
That worked, thank you

**Reply by Koenkk** [reply]
Could you provide the debug log when triggering it in 1.40.1?

See [this](https://www.zigbee2mqtt.io/guide/configuration/logging.html) on how to enable debug logging.

**Comment by Claude2666** [comment]
I rolled back to 1.40.1-dev commit: [0540e4fb]

Attached a debug log of my tilt sensor, I hope it contains the needed info.

[tilt.txt](https://github.com/user-attachments/files/17274574/tilt.txt)

**Reply by MarcoZ76** [reply]
Can you tell me how to downgrade to the previous version? I have HA OS and the Z2M is installed as addon component. Thanks

**Reply by Maggie70** [reply]
when updating created a backup of the old one, read it back when it didn't work with the new one.

**Comment by raforo** [comment]
I have the same issue in two Aqara DJT11LM, downgrading to previous version works as usual. 
Shaking the device, shows vibration and action changes, but angles remain unknown.

**Comment by rojoricardo** [comment]
I'm having the same issue with latest version, I'm downgrading right now, and I'll check and update.

**UPDATE** I can confirm that performing the downgrade / Restoring backup of version 1.40.1-1 fixed the issue.

**Comment by Koenkk** [comment]
fixed!

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Reply by MarcoZ76** [reply]
Thanks! When will be available on stable release?

**Reply by Koenkk** [reply]
1 November

**Reply by Nevynnyi** [reply]
Thx!!! ☺️

**Comment by jeremysherriff** [comment]
Confirmed fixed using docker tag `latest-dev` with zigbee-herdsman-converters release `20.25.0`, many thanks

Note to others: the values seem to stay 0/null until I made the device sends through a vibration event, at which time the values all updated.

**Comment by paulbrause** [comment]
Updated to latest Dev-Version, but still not working correctly. Values updating one time after first vibration event, but after that no further update of the axes and accelerometer values.

**Reply by jeremysherriff** [reply]
> Updated to latest Dev-Version, but still not working correctly. Values updating one time after first vibration event, but after that no further update of the axes and accelerometer values.

~~Agreed, I'm seeing the same behaviour.  Things are not working right, @Koenkk. I think this needs to be reopened.~~

@paulbrause I've had more time to experiment today and I cannot reproduce an issue on `1.40.2-dev commit: db00759` with zigbee-herdsman-converters release `20.25.0`. To me this looks to be resolved, I'm not sure what was causing the data to not update last night but today I cannot fault it.

Edit: If it helps (?) I rolled back to 1.40.1, and then forward to latest-dev. Having checked the edit in the commit I cannot see why this would make a difference, but it is what I did.

**Reply by paulbrause** [reply]
> > Updated to latest Dev-Version, but still not working correctly. Values updating one time after first vibration event, but after that no further update of the axes and accelerometer values.
> 
> ~Agreed, I'm seeing the same behaviour. Things are not working right, @Koenkk. I think this needs to be reopened.~
> 
> @paulbrause I've had more time to experiment today and I cannot reproduce an issue on `1.40.2-dev commit: db00759` with zigbee-herdsman-converters release `20.25.0`. To me this looks to be resolved, I'm not sure what was causing the data to not update last night but today I cannot fault it.
> 
> Edit: If it helps (?) I rolled back to 1.40.1, and then forward to latest-dev. Having checked the edit in the commit I cannot see why this would make a difference, but it is what I did.

@jeremysherriff You are right. After rolling back to 1.40.1. and then updating to latest-dev everything works fine. Thank you!

**Comment by jeremysherriff** [comment]
Keen to hear from others on whether this issue is fully resolved.

My vibration sensor seems to be very unreliable reporting the x/y/z angles and axis values to HA now, but when I set z2m logging to info and filter on the device, the values appear to be recorded in a timely manner and be accurate.  Have changed to a fresh battery but the old one was still above 3000mV so it should have been fine, and the fresh one has only helped for a few days (which was one of the original symptoms reported, despite the code fix being to do with the data payload parsing).

What are other people's experiences?
Q also posted in the still-open Issue #21432

**Reply by rojoricardo** [reply]
My sensors are running through batteries like crazy too, I thought I had bought falty batch of batteries since I ordered them online, but looking at your post, I'm starting to think it might be something else.

Is anyone else experiencing something like this?

**Reply by Claude2666** [reply]
In my case. the problem was solved. The DJT11LM sensor behaves as it always had.  One observation is that the angle reporting is not fast, it takes a while. This is has always been like that in my case. 
And in general, my experience with Xiaomi Zigbee sensors is that the battery level and voltage reporting is not very reliable. They always stay 100%, and suddenly stop responding because the battery is dead.  
I also had some Xiaomi sensors that went through batteries very fast. It always started after I replaced the battery. I then replaced the sensors, but I suspect something else is going on. Since I replaced some of the old Zigbee 1.2 only routing devices, I have not seen any battery drain anymore. Could be coincidence...

**Comment by Nevynnyi** [comment]
Broke again. Zigbee2MQTT v 2.1.1

Angle x N/A°
Angle y N/A°
Angle z N/A°

X axis N/A
Y axis N/A
Z axis N/A

and not update

**Reply by MurilloRacing** [reply]
I had to back date to 1.41.0-1 to get X Y Z axis data working again. I tried updating and back dating few times The only way I was able to maintain stable reliable X Y Z axis data was to use 1.41.0-1.

**Reply by florianmulatz** [reply]
Seems buggy for mee too. Sometimes it detects changes - sometimes not. I can not rely on that - any thing I can help with?

Cheers

PS: v2.2.1 - I don't downgrade it since this is my first vibration-sensor and I don't want to risk to break other things

**Comment by Nevynnyi** [comment]
Still not work on z2m 2.3.0

**Comment by jeremysherriff** [comment]
Working perfectly here. I had to make some changes though; 

-  good quality battery
-  unpaired and re-paired directly to coordinator only

Hope that helps

**Comment by dingausmwald** [comment]
EEEEVERY UPDATE SOMETHING BREAKS. EEEEEEEEVERY UPDATE. I DONT EVEN UPDATE ANYMORE, BUT WHEN I DO, THE WHOLE DAY IS GONE BECAUSE OF ZIGBEE2MQTT.

Angles still dont report reliably.
